scipy/stats/_distn_infrastructure.py

```diff
@@ -1426,8 +1426,26 @@ def support(self, *args, **kwargs):
 
         """
         args, loc, scale = self._parse_args(*args, **kwargs)
-        _a, _b = self._get_support(*args)
-        return _a * scale + loc, _b * scale + loc
+        arrs = np.broadcast_arrays(*args, loc, scale)
+        args, loc, scale = arrs[:-2], arrs[-2], arrs[-1]
+        cond = self._argcheck(*args) & (scale > 0)
+        if cond.all():
+            _a, _b = self._get_support(*args)
+            return _a * scale + loc, _b * scale + loc
+        goodargs = argsreduce(cond, loc, scale, *args)
+        goodloc, goodscale = goodargs[:2]
+        goodargs = goodargs[2:]
+        _a, _b = self._get_support(*goodargs)
+        out_a = np.zeros(loc.shape, dtype='d')
+        out_b = np.zeros(loc.shape, dtype='d')
+        out_a_, out_b_ = _a * goodscale + goodloc, _b * goodscale + goodloc
+        place(out_a, cond, out_a_)
+        place(out_a, 1-cond, self.badvalue)
```

Comment:
```suggestion
        place(out_a, ~cond, self.badvalue)
```
Asking other developers - is this, or `np.logical_not`, not preferable?

Reply:
The `_argcheck` method returns `int` arrays/scalars for some distributions while `bool` arrays/scalars for others. For example,

```python
>>> from numpy import array
>>> from scipy.stats import norm, t
>>> norm._argcheck().__class__
<class 'int'>
>>> t._argcheck(1).__class__
<class 'bool'>
```

When it returns an integer, `~` would perform ones complement rather than logical not, hence, returning wrong `cond` array/scalar. To avoid that, I guess, `1-cond` is present rather than `~cond`. On the other hand, your suggestion about using a call to `np.logical_not` is valid. It will convert the arrays to `bool` and then perform a logical not. Though, I don't know if it would be any better or not :sweat_smile:

Reply:
Probably not more efficient or anything. I was asking as a matter of style.